main.go

```diff
@@ -118,8 +169,40 @@ func main() {
 		}
 	})
 
-	// send klog messages through our zap logger so they look the same (console vs JSON)
-	klog.SetLogger(logr)
+	// send klog messages through a zap logger so they have the same format
+	if zapEncoder == "console" { // klog already adds a newline, so have zap skip adding one.
+		zapcfg.EncoderConfig.SkipLineEnding = true
+	}
+
+	klogLevel, err := strconv.Atoi(pflag.Lookup("v").Value.String())
+	if err != nil {
+		log.Error(err, "Invalid value passed to 'v' flag - using '0' as a default")
+
+		klogLevel = 0
+	}
+
+	zapcfg.Level = zap.NewAtomicLevelAt(zapcore.Level(int8(-1 * klogLevel)))
+
+	kZapLog, err := zapcfg.Build()
+	if err != nil {
+		panic(fmt.Sprintf("Failed to build zap Logger for klog from configuration: %v", err))
+	}
+
+	klog.SetLogger(zapr.NewLogger(kZapLog).WithName("klog"))
+
+	klog.V(0).Info("klog 0")
+	klog.V(1).Info("klog 1")
+	klog.V(2).Info("klog 2")
+	klog.V(3).Info("klog 3")
+	klog.V(4).Info("klog 4")
+	klog.V(5).Info("klog 5")
```

Comment:
Running with `--zap-log-level=1 --zap-encoder=json --v=-2`, I see:
```
{"level":"info","ts":"2022-03-11T17:24:34.942-0500","logger":"setup","caller":"logr@v1.2.2logr.go:252","msg":"zap 0"}
{"level":"debug","ts":"2022-03-11T17:24:34.942-0500","logger":"setup","caller":"logr@v1.2.2logr.go:252","msg":"zap 1"}
```

With `--zap-log-level=4 --v=1`, I see:
```
2022-03-11T17:25:00.961-0500	info	klog	config-policy-controllermain.go:193	klog 0
2022-03-11T17:25:00.961-0500	debug	klog	config-policy-controllermain.go:194	klog 1
2022-03-11T17:25:00.961-0500	info	setup	logr@v1.2.2logr.go:252	zap 0
2022-03-11T17:25:00.961-0500	debug	setup	logr@v1.2.2logr.go:252	zap 1
2022-03-11T17:25:00.961-0500	lvl-2	setup	logr@v1.2.2logr.go:252	zap 2
2022-03-11T17:25:00.961-0500	lvl-3	setup	logr@v1.2.2logr.go:252	zap 3
2022-03-11T17:25:00.961-0500	lvl-4	setup	logr@v1.2.2logr.go:252	zap 4
```

And finally, with `--zap-log-level=1 --v=3`, I see:
```
2022-03-11T17:26:28.795-0500	info	klog	config-policy-controllermain.go:193	klog 0
2022-03-11T17:26:28.795-0500	debug	klog	config-policy-controllermain.go:194	klog 1
2022-03-11T17:26:28.795-0500	lvl-2	klog	config-policy-controllermain.go:195	klog 2
2022-03-11T17:26:28.795-0500	lvl-3	klog	config-policy-controllermain.go:196	klog 3
2022-03-11T17:26:28.795-0500	info	setup	logr@v1.2.2logr.go:252	zap 0
2022-03-11T17:26:28.795-0500	debug	setup	logr@v1.2.2logr.go:252	zap 1
```

So things are mostly working. The main thing to fix is the file that zap is reporting (those logs show they're coming from logr).